system-contracts/contracts/precompiles/EcPairing.yul

```diff
@@ -4,149 +4,70 @@ object "EcPairing" {
-            /// @dev E': y' ** 2 = x' ** 3 + 3 / (9 + u)
-            /// @dev the curve E' is defined over Fp2 elements.
-            /// @dev See https://hackmd.io/@jpw/bn254#Twists for further details.
-            /// @return coefficients of the sextic twist of the BN256 curve
-            function MONTGOMERY_TWISTED_CURVE_COEFFS() -> z0, z1 {
-                z0 := 16772280239760917788496391897731603718812008455956943122563801666366297604776
-                z1 := 568440292453150825972223760836185707764922522371208948902804025364325400423
-            }
-
-            /// @notice Constant function for the alt_bn128 group order.
-            /// @dev See https://eips.ethereum.org/EIPS/eip-196 for further details.
-            /// @return ret The alt_bn128 group order.
-            function P() -> ret {
-                ret := 21888242871839275222246405745257275088696311157297823662689037894645226208583
-            }
-
-            /// @notice Constant function for the twisted curve subgroup order.
-            /// @dev See https://hackmd.io/@jpw/bn254#Parameter-for-BN254 for further details.
-            /// @return ret The twisted curve subgroup orde.
-            function TWISTED_SUBGROUP_ORDER() -> ret {
-                ret := 21888242871839275222246405745257275088548364400416034343698204186575808495617
-            }
-
-            /// @notice Constant function for the pre-computation of R^2 % N for the Montgomery REDC algorithm.
-            /// @dev R^2 is the Montgomery residue of the value 2^512.
-            /// @dev See https://en.wikipedia.org/wiki/Montgomery_modular_multiplication#The_REDC_algorithm for further details.
-            /// @dev This value was precomputed using Python.
-            /// @return ret The value R^2 modulus the curve group order.
-            function R2_MOD_P() -> ret {
-                ret := 3096616502983703923843567936837374451735540968419076528771170197431451843209
-            }
-
-            /// @notice Constant function for the pre-computation of N' for the Montgomery REDC algorithm.
-            /// @dev N' is a value such that NN' = -1 mod R, with N being the curve group order.
-            /// @dev See https://en.wikipedia.org/wiki/Montgomery_modular_multiplication#The_REDC_algorithm for further details.
-            /// @dev This value was precomputed using Python.
-            /// @return ret The value N'.
-            function N_PRIME() -> ret {
-                ret := 111032442853175714102588374283752698368366046808579839647964533820976443843465
+    object "EcPairing_deployed" {
+        code {
+            ////////////////////////////////////////////////////////////////
+            //                      CONSTANTS
+            ////////////////////////////////////////////////////////////////
+
+            /// @dev The basic gas cost of processing ecpairing circuit precompile.
+            function ECPAIRING_BASE_GAS_COST() -> ret {
```

Comment:
Very interesting, what will happen if the input is empty bytes? The number of pairs - 0, but the circuit will need to be called, no?